S2Geometry/DataStructures/NullObject.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Google.Common.Geometry.DataStructures
+{
+    public struct NullObject<T>
+    {
+        [DefaultValue(true)]
+        private bool isnull;// default property initializers are not supported for structs
```

Comment:
Does this actually work? If not, how about inverting it to default as `bool isNotNull`. I know it's a double-negative, but it plays better with default struct semantics.